### PR TITLE
Format extensions and get rid of deprecated Rank2Types extension

### PR DESCRIPTION
**Pull request by lehins**
No semantic changes, just minor syntactic improvements that will reduce git diff in the future

### CODE REVIEW

File: vector/src/Data/Vector.hs [hunk 1]
```diff
@@ -1,11 +1,10 @@
...
-           , Rank2Types
-           , BangPatterns
-  #-}
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE BangPatterns #-}
```
Comment (konsumlamm) [inline vector/src/Data/Vector.hs:7]:
```suggestion
{-# LANGUAGE BangPatterns #-}
{-# LANGUAGE CPP #-}
{-# LANGUAGE DeriveDataTypeable #-}
{-# LANGUAGE FlexibleInstances #-}
{-# LANGUAGE MultiParamTypeClasses #-}
{-# LANGUAGE RankNTypes #-}
{-# LANGUAGE TypeFamilies #-}
```
We might as well sort them alphabetically, while we're at it (that also seems to have been done for the other modules).

File: vector/src/Data/Vector/Generic/New.hs [hunk 2]
```diff
@@ -1,5 +1,7 @@
-{-# LANGUAGE CPP, Rank2Types, FlexibleContexts, MultiParamTypeClasses #-}
-
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
```
Comment (konsumlamm) [inline vector/src/Data/Vector/Generic/New.hs:4]:
```suggestion
{-# LANGUAGE CPP #-}
{-# LANGUAGE FlexibleContexts #-}
{-# LANGUAGE MultiParamTypeClasses #-}
{-# LANGUAGE RankNTypes #-}
```

**Comment by lehins** [comment]
@konsumlamm Thanks. I thought I did sort all of them, but I guess I missed a couple modules.

**Comment by lehins** [comment]
Dammit, meant to squash the commits. Oh well.

**Pull request merged**